atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraKeyValueServices.java

```diff
@@ -88,19 +88,18 @@ static void waitForSchemaVersions(
         long sleepTime = INITIAL_SLEEP_TIME;
         Map<String, List<String>> versions;
         do {
-            // This only returns the schema versions of nodes that the client knows exist. In particular, if a node we
-            // shook hands with goes down, it will have schema version UNREACHABLE; however, if we never shook hands
-            // with a node, there will simply be no entry for it in the map. Hence the check for the number of nodes.
+            // This may only include some of the nodes if the coordinator hasn't shaken hands with someone; however,
+            // this existed largely as a defense against performance issues with concurrent schema modifications.
             versions = client.describe_schema_versions();
-            if (uniqueSchemaWithQuorumAgreementAndOtherNodesUnreachable(config, versions)) {
+            if (reachablePartOfClusterHasConsistentSchemaVersions(versions)) {
                 return;
             }
             sleepTime = sleepAndGetNextBackoffTime(sleepTime);
         } while (System.currentTimeMillis() < start + config.schemaMutationTimeoutMillis());
 
         StringBuilder schemaVersions = new StringBuilder();
         for (Entry<String, List<String>> version : versions.entrySet()) {
-            schemaVersions = addNodeInformation(schemaVersions,
+            addNodeInformation(schemaVersions,
                     String.format("%nAt schema version %s:", version.getKey()),
                     version.getValue());
         }
```

Comment:
all of the error message stuff below should talk about the cluster as being in terms of node returned in the describe_schema_versions() call above (regardless of if they're unreachable or not). none of it should use `config.servers()` as that isn't correct and also just isn't even how the checks of the rest of the method are done anymore.

Reply:
Yep, now updated to refer to the output returned from the describe_schema_versions call